platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java

```diff
@@ -338,6 +336,56 @@ private static Path getConsumingProjectIdentityPath(ResolveContext resolveContex
         return null;
     }
 
+    /**
+     * Perform dependency resolution and visit the results.
+     */
+    private void doResolve(
+        ResolveContext resolveContext,
+        ComponentResolvers resolvers,
+        boolean includeSyntheticDependencies,
+        Spec<DependencyMetadata> edgeFilter,
+        ImmutableList<DependencyGraphVisitor> visitors
+    ) {
+        ResolutionStrategyInternal resolutionStrategy = resolveContext.getResolutionStrategy();
+
+        if (resolutionStrategy.isDependencyLockingEnabled()) {
+            if (resolutionStrategy.isFailingOnDynamicVersions()) {
+                throw new InvalidUserCodeException(
+                    "Resolution strategy has both dependency locking and fail on dynamic versions enabled. " +
+                        "You must choose between the two modes."
+                );
+            } else if (resolutionStrategy.isFailingOnChangingVersions()) {
+                throw new InvalidUserCodeException(
+                    "Resolution strategy has both dependency locking and fail on changing versions enabled. " +
+                        "You must choose between the two modes."
+                );
+            }
+        }
+
+        // TODO: These dependencies should not be provided separately, but should be part of the root variant.
+        List<? extends DependencyMetadata> syntheticDependencies = includeSyntheticDependencies ?
+            resolveContext.getSyntheticDependencies() : Collections.emptyList();
+
+        // ResolveContext and ResolutionStrategy should not be passed into the dependency graph resolver.
+        // The resolver should be agnostic of these types so that it may be used to resolve
+        // non-Configuration types.
```

Comment:
💭 I'm confused about what this comment is for. These aren't passed in, are you worried people will try to refactor this to pass them in the future? If so, I would rather have this comment at the declaration site of `DependencyGraphResolver#resolve`.

Reply:
You are correct. 

Moving this to `DependencyGraphResolver#resolve`